public/components/visualization/text2viz.tsx

```diff
@@ -142,7 +144,9 @@ export const Text2Viz = () => {
     const subscription = text2vega?.getResult$().subscribe((result) => {
       if (result) {
         if (result.error) {
-          notifications.toasts.addError(result.error, {
+          const msg = `Unable to generate a visualization. ${result.error.message}`;
+          setErrorMessage(msg);
```

Comment:
Nit: call `setErrorMessage(msg);` in line 156, though we will reset the error message at very beginning of onSubmit. I just feel it more safe to handle the error message in all the branches.